account_financial_report_webkit/models/account_static_balance.py

```diff
@@ -253,3 +291,31 @@ def map_data_to_account_id(data):
                                          entry['curr_balance']),
                     })
         return dict(item for item in res.iteritems() if item[0] in return_ids)
+
+    def _validate_journal_integrity(self, period, journal):
+        """ Validate if a journal is still valid (not outdated). """
+        account_period = self.env['account.journal.period'].search([
+            ('journal_id', '=', journal.id),
+            ('period_id', '=', period.id),
+        ])
+
+        if not account_period:
+            return False
+
+        static_balances = self.env['account.static.balance'].search([
+            ('journal_id', '=', journal.id),
+            ('period_id', '=', period.id),
+        ])
+
+        if not static_balances:
+            return False
+
+        min_date = max(static_balances.mapped('create_date'))
+        if account_period.write_date > min_date:
+            logger.info(
+                'Integrity of static balance period %s with journal %s is not '
+                'guranteed', period.id, journal.id)
+            static_balances.unlink()
```

Comment:
By this unlink, there is an immediately recalculation triggered.

Reply:
You can safely unlink when calculate_balances is called to store the new balances in the database, but not when it's called for report generation because this will cause concurrency errors in the case of two simulaneous reports (which is not uncommon). I would do this in or before 'get_missing_entries' so that they show up as missing entries automatically. A database query across the full table would be the quickest, something like 
```
delete from account_static_balance asb
where exists (select * from account_journal_period ajp
              where ajp.period_id = asb.period_id AND ajp.journal_id = asb.period_id
                   and asb.create_date <= ajp.write_date or ajp.state != 'done')
```

Reply:
Used your query in a seperate method. And triggered the method in`calculate_static_balances` and `get_missing_entries` because get_missing_entries is triggered by the balance generation (and daily cronjob), and calculate_static_balances is used when closing a journal_period